pcu/PCU.cc

```diff
@@ -232,9 +258,15 @@ PCU &PCU::operator=(PCU && other) noexcept {
   std::swap(msg_, other.msg_);
   return *this;
 }
-MPI_Comm PCU::GetMPIComm() const noexcept { return mpi_->original_comm; }
+PCU_Comm PCU::GetMPIComm() const noexcept { return GetComm(); }
+PCU_Comm PCU::GetComm() const noexcept { return mpi_->original_comm; }
```

Comment:
I'm wondering if we should remove this from the interface. Otherwise, it may be too tempting to drop out of `PCU` to `MPI` which will break the no mpi use case.

Reply:
I think there are cases where users may want access to advanced MPI features. A simple one I can think of (I have limited MPI experience) is if a user wants to call MPI_Comm_compare.

Reply:
I think we should keep it.  Maybe adding a docstring with the suggestion that collective and p2p messaging that involves pumi objects should use PCU interfaces and not direct MPI calls.